src/Compilers/Core/Portable/Emit/CommonPEModuleBuilder.cs

```diff
@@ -89,6 +89,23 @@ public CommonPEModuleBuilder(
         /// EnC generation. 0 if the module is not an EnC delta, 1 if it is the first EnC delta, etc.
         /// </summary>
         public int CurrentGenerationOrdinal => (PreviousGeneration?.Ordinal + 1) ?? 0;
+
+        /// <summary>
+        /// Creates the type definition of HotReloadException type if it has not been synthesized yet and returns its constructor.
+        /// </summary>
+        public abstract IMethodSymbolInternal GetOrCreateHotReloadExceptionConstructorDefinition();
+
+        /// <summary>
+        /// Creates the type definition of HotReloadException type if it has not been synthesized yet and the module is an EnC delta.
+        /// Returns the synthesized type definition or null if the module is not an EnC delta or a user-defined type is already defined in the compilation.
```

Comment:
>a user-defined type is already defined in the compilation [](http://example.com/codeflow?start=97&length=57)

If a user-defined type is already defined, doesn't this method return it rather than `null`?